### DISCUSSION TITLE
Dynamic imports vs. static builds

### DISCUSSION THREAD

**Discussion by jpommerening** [matteobruni/tsparticles#5323] [Q&A]
Hi!

I had a lot of fun playing with tsParticles the past few days :)

However, now I'm starting to bundle my setup for production and I'm having difficulties with the dynamic imports.

I'm using Rollup via Vite with no custom configuration (yet).
My code is importing from `@tsparticles/engine`, `@tsparticles/basic` and `@tsparticles/interaction-particles-links`:

```typescript
import {colorToHsl, getHslAnimationFromHsl, ISourceOptions, tsParticles} from "@tsparticles/engine";
import {loadBasic} from "@tsparticles/basic";
import {loadParticlesLinksInteraction} from "@tsparticles/interaction-particles-links";
```

Rollup generates output files for all the internal dynamic imports (which I assume is according to plan):
```
dist/assets/plugin-Bvm9Cs6i.js                0.47 kB │ gzip:   0.28 kB
dist/assets/index-B-EbEdZc.js                 0.47 kB │ gzip:   0.28 kB
dist/assets/CircleDrawer-B-oQI5ug.js          0.48 kB │ gzip:   0.34 kB
dist/assets/interaction-D9Zh3V_y.js           0.49 kB │ gzip:   0.29 kB
dist/assets/index-sUgSMNns.js                 0.50 kB │ gzip:   0.29 kB
dist/assets/index-CHfiUF0U.js                 0.50 kB │ gzip:   0.29 kB
dist/assets/index-TbxKB1xP.js                 0.51 kB │ gzip:   0.29 kB
dist/assets/LinksPlugin-DbtSqsQt.js           0.52 kB │ gzip:   0.31 kB
dist/assets/ColorUpdater-OJOECZp8.js          0.54 kB │ gzip:   0.32 kB
dist/assets/SizeUpdater-H9AYPxjw.js           0.56 kB │ gzip:   0.33 kB
dist/assets/OpacityUpdater-xIHhyX_h.js        0.67 kB │ gzip:   0.36 kB
dist/assets/index-DQ7AuT4f.js                 1.34 kB │ gzip:   0.65 kB
dist/assets/Linker-Cm-NZ83n.js                2.34 kB │ gzip:   1.03 kB
dist/assets/BaseMover-DhpREy-h.js             3.27 kB │ gzip:   1.42 kB
dist/assets/LinkInstance-CGWFW30I.js          4.05 kB │ gzip:   1.67 kB
dist/assets/OutOfCanvasUpdater-sQlAty6w.js    6.12 kB │ gzip:   1.73 kB
dist/assets/Container-DSBVu-kY.js            36.20 kB │ gzip:  11.04 kB
```

Do you know if there's a way to statically package what's needed into a single bundle, to not have so many HTTP requests on initial load?
Is something like that planned, maybe?

I know that the exact modules that are required depend on the options that I pass to tsParticles, which is probably why the dynamic imports were added in the first place, as to not load everything at once.
So I probably have to live with some unnecessary code.
But is it possible at all?
I tried Rollup's `inlineDynamicImports` option, but then I can't have any chunks at all, which is not so nice either…

Thanks :)

**Comment by matteobruni** [comment]
I'm reworking that for 3.4.0, with the CDN is fast, because of cache, but with custom builds this could not be true. I could take time to fix this, if you don't need some of the 3.3.0, you can go back to 3.2.x version, that is not using dynamic imports
